crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -396,6 +404,49 @@ impl EventCacheInner {
                 continue;
             };
 
+            // Handle and cache events and relations
+            let mut relations = self.relations.write().await;
+            for ev in &joined_room_update.timeline.events {
+                if let Ok(AnySyncTimelineEvent::MessageLike(ev)) = ev.event.deserialize() {
+                    if let Some(AnyMessageLikeEventContent::RoomMessage(c)) = ev.original_content()
+                    {
+                        if let Some(Relation::Replacement(relation)) = c.relates_to {
+                            if let Some(replacements) = relations.get_mut(&relation.event_id) {
+                                replacements.insert(ev.event_id().to_owned());
+                            } else {
+                                relations.insert(
+                                    relation.event_id,
+                                    BTreeSet::from_iter(vec![ev.event_id().to_owned()]),
+                                );
+                            }
+                        }
+                    } else if let Some(AnyMessageLikeEventContent::RoomRedaction(c)) =
+                        ev.original_content()
+                    {
+                        if let Some(redacted_ev_id) = c.redacts {
```

Comment:
It's optional because in previous versions than v11, the redacts field was somewhere else, if I remember correctly (in the event itself, not the content). Can you handle that case too?

Reply:
So... this seems way more difficult than expected. In the `EventCache` I have no way to access the `RoomBaseInfo` to check the room version.

Reply:
Maybe I can try not relying on the version id and just checking the event: if the redaction is in the event itself, use it, otherwise, check for the redaction inside the contents.